src/AzureAppConfigurationImpl.ts

```diff
@@ -105,3 +108,23 @@ export class AzureAppConfigurationImpl extends Map<string, unknown> implements A
         return headers;
     }
 }
+
+function validatedSelectors(selectors?: { keyFilter: string, labelFilter?: string }[]) {
+    if (!selectors || selectors.length === 0) {
+        // Default selector: key: *, label: \0
+        return [{ keyFilter: KeyFilter.Any, labelFilter: LabelFilter.Null }];
+    }
+    return selectors.map(selectorCandidate => {

```

Comment:
Will this return unique selectors if there are any duplicates?

Reply:
No, but the settings are ensured to be unique, as they are overridden by the last selector if there are any duplicates or overlaps among selectors.

We cannot identify overlaped selectors and combine them into one, but yes we can dedup the selectors with exactly the same keyFilter and labelFilter, to save requests (if they do specify the same selectors multiple times).

Reply:
> but yes we can dedup the selectors with exactly the same keyFilter and labelFilter, to save requests (if they do specify the same selectors multiple times).

Yes, that's the case I was talking about. Dedup exactly same key and label filters to save requests.

Reply:
@avanigupta I have a question. Since the order of selectors also reflects priority, what would be the effective selector for this case?
I think current impl in .NET provider has the same problem.
```ts
selectors: [ 
            { keyFilter: "app.*", labelFilter: "dev" },
            { keyFilter: "app.*", labelFilter: "production" },
            { keyFilter: "app.*", labelFilter: "dev" },
] 
```

Without dedup, it's clear that the last "app.* + dev" wins. But with simple dedup, the last selector is not counted in, the effective one becomes  "app.* + production". I think current impl in .NET provider has the same problem.

Do you think we should reverse the array before dedup?

Reply:
That's a good catch. This looks like a [bug in the .NET provider](https://github.com/Azure/AppConfiguration-DotnetProvider/blob/75f3d6c186bbfb8c43e62f37207b1193e66e39e6/src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs#L154C27-L154C27). We need to deduplicate while maintaining the priority of the last occurrence. In this case, "app.* + dev" should always win.

cc @jimmyca15 @amerjusupovic

Reply:
Shall I create a separate issue in .NET provider repo?

Reply:
> Shall I also create a separate issue in .NET provider repo?

Sounds like a good idea.

Reply:
@avanigupta 
Here is the issue for .NET provider: https://github.com/Azure/AppConfiguration-DotnetProvider/issues/490

BTW, I also created an issue here in JS provider, and we can add the dedup logic correctly later in a separate PR dedicated for that purpose. For this PR, I would like to get it merged as it is since the behavior is as expected. WDYT?